olive/systems/azureml/aml_system.py

```diff
@@ -130,79 +146,79 @@ def run_pass(
 
             return self._load_model(model, output_model_path, pipeline_output_path)
 
-    def _create_model_inputs(self, model_path_type: ResourceType):
-        # model_path_type can be None if the model_path is None
-        # we create a placeholder input for model_path in this case
-        model_path_asset_type = RESOURCE_TYPE_TO_ASSET_TYPE[model_path_type] if model_path_type else AssetTypes.URI_FILE
+    def _create_model_inputs(self, model_path_type: AssetTypes):
         return {
             "model_config": Input(type=AssetTypes.URI_FILE),
-            "model_path": Input(type=model_path_asset_type, optional=True),
+            "model_path": Input(type=model_path_type, optional=True),
             "model_script": Input(type=AssetTypes.URI_FILE, optional=True),
             "model_script_dir": Input(type=AssetTypes.URI_FOLDER, optional=True),
         }
 
-    def _create_model_args(self, model_json: dict, tmp_dir: Path):
-        model_script = None
-        if model_json["config"].get("model_script"):
-            model_script = Input(type=AssetTypes.URI_FILE, path=model_json["config"]["model_script"])
-            model_json["config"]["model_script"] = None
+    def _get_path_from_resource_path(self, rp: OLIVE_RESOURCE_ANNOTATIONS):
+        model_resource_path = create_resource_path(rp)
+        asset_type = get_asset_type_from_resource_path(model_resource_path)
+
+        if model_resource_path.type == ResourceType.AzureMLDatastore:
+            asset_type = AssetTypes.URI_FILE if model_resource_path.is_file() else AssetTypes.URI_FOLDER
+
+        if model_resource_path.type in AZUREML_RESOURCE_TYPES:
+            # ensure that the model is in the same workspace as the system
+            model_workspace_config = model_resource_path.get_aml_client_config().get_workspace_config()
+            system_workspace_config = self.azureml_client_config.get_workspace_config()
+            for key in model_workspace_config:
+                if model_workspace_config[key] != system_workspace_config[key]:
+                    logger.warning(
+                        f"Model workspace {model_workspace_config} is different from system workspace"
+                        f" {system_workspace_config}. Olive will download the model to local storage, then upload it to"
+                        "the system workspace."
+                    )
+                    return Input(type=asset_type, path=get_local_path(model_resource_path))
```

Comment:
>get_local_path [](http://example.com/codeflow?start=55&length=14)

for azureml scenario, do we need download the model locally and upload it again for pipelien jobs?